### PR TITLE
Extend the simulator to a linear sequence of tasks

### PR DESCRIPTION
**Pull request by MaoZiming**

### CODE REVIEW

**Comment by MaoZiming** [comment]
Kept the original `type.py`.

File: ray_data_eval/common/pipeline.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+                for i in range(operator.num_tasks)
+            ]
+        )
+    return tasks
+
+
+@dataclass
+class SchedulingProblem:
+    operator_list: list[OperatorSpec]
+    name: str
```
Comment (franklsf95) [inline ray_data_eval/common/pipeline.py:48]: add `tasks: list = field(init=False)`

File: ray_data_eval/common/pipeline.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    operator_idx: int
+    duration: int
+    input_size: int
+    output_size: int
+    num_cpus: int
+
+
+def _get_tasks(operator_list: list[OperatorSpec]):
+    tasks = []
+    # Reversed so that downstream tasks are prioritized.
```
Comment (franklsf95) [inline ray_data_eval/common/pipeline.py:27]: This design looks a little brittle; I don't think the policies should rely on this ordering. It's fine to keep as is, but we should probably sort again in the policies.

File: ray_data_eval/simulator/benchmark.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+from ray_data_eval.simulator.environment import ExecutionEnvironment
+from ray_data_eval.simulator.policies import (  # noqa F401
+    GreedySchedulingPolicy,
+    GreedyWithBufferSchedulingPolicy,
+    GreedyAndAnticipatingSchedulingPolicy,
+    SchedulingPolicy,
+    RatesEqualizingSchedulingPolicy,
+)
+
+logging.disable(logging.CRITICAL)
```
Comment (franklsf95) [inline ray_data_eval/simulator/benchmark.py:12]: uh, why?

**Review by franklsf95 — APPROVED** [review]
Great job!!

File: ray_data_eval/common/pipeline.py (hunk 2 above)
Comment (MaoZiming) [reply ray_data_eval/common/pipeline.py:27]: Added a todo: I think eventually the implementation of the policy should not rely on ordering.

File: ray_data_eval/simulator/benchmark.py (hunk 3 above)
Comment (MaoZiming) [reply ray_data_eval/simulator/benchmark.py:12]: I am thinking about keeping the output clean:
```
Problem: test_problem
GreedySchedulingPolicy Finished 6
GreedyWithBufferSchedulingPolicy Finished 6
GreedyAndAnticipatingSchedulingPolicy Finished 6
RatesEqualizingSchedulingPolicy Finished 6
---
Problem: multi_stage_problem
GreedySchedulingPolicy Finished 12
GreedyWithBufferSchedulingPolicy Finished 12
GreedyAndAnticipatingSchedulingPolicy Finished 12
RatesEqualizingSchedulingPolicy Finished 11
---
Problem: producer_consumer_problem
GreedySchedulingPolicy Finished 12
GreedyWithBufferSchedulingPolicy Finished 12
GreedyAndAnticipatingSchedulingPolicy Finished 12
RatesEqualizingSchedulingPolicy Finished 12
---
Problem: long_problem
GreedySchedulingPolicy Finished 60
GreedyWithBufferSchedulingPolicy Finished 60
GreedyAndAnticipatingSchedulingPolicy Finished 60
RatesEqualizingSchedulingPolicy Finished 59
---
```

Comment (MaoZiming) [reply ray_data_eval/simulator/benchmark.py:12]: removed 
`logging.disable(logging.CRITICAL)`

**Pull request merged**